### PR TITLE
Fix wrong table name

### PR DESCRIPTION
**Pull request by lunny**
The table name should be `oauth2_application` but `o_auth2_application`

Caused by https://github.com/go-gitea/gitea/pull/21316/files#diff-9610efbc608a41f1f2eaff5790423f0a187906f6ff0beb23a5e8d18366cc2ccfR38

### CODE REVIEW

**Comment by lunny** [comment]
> ```
> func AddConfidentialClientColumnToOAuth2ApplicationTable(x *xorm.Engine) error {
> 	type OAuth2Application struct {
> 		ConfidentialClient bool `xorm:"NOT NULL DEFAULT TRUE"`
> 	}
> 
> 	return x.Sync(new(OAuth2Application))
> }
> ```
> 
> The real bug is here. It doesn't provide a TableName method.

Fixed.

**Review by silverwind — APPROVED** [review]

File: models/migrations/v1_18/v230_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func Test_AddConfidentialClientColumnToOAuth2ApplicationTable(t *testing.T) {
 	}
 
 	got := []ExpectedOAuth2Application{}
-	if err := x.Table("o_auth2_application").Select("id, confidential_client").Find(&got); !assert.NoError(t, err) {
+	if err := x.Table("oauth2_application").Select("id, confidential_client").Find(&got); !assert.NoError(t, err) {
```
Comment (techknowlogick) [inline models/migrations/v1_18/v230_test.go:39]: is a new migration needed to rename the table if it is incorrectly named?

Comment (lunny) [reply models/migrations/v1_18/v230_test.go:39]: Now the database should have both tables. The `oauth2_application` is the correct one and `o_auth2_application` is the incorrect.

Comment (silverwind) [reply models/migrations/v1_18/v230_test.go:39]: Maybe migration to drop the incorrect table?

Comment (lunny) [reply models/migrations/v1_18/v230_test.go:39]: f8d57dc

File: models/migrations/v1_18/v230.go [hunk 2]
```diff
@@ -10,8 +10,9 @@ import (
 // AddConfidentialColumnToOAuth2ApplicationTable: add ConfidentialClient column, setting existing rows to true
 func AddConfidentialClientColumnToOAuth2ApplicationTable(x *xorm.Engine) error {
 	type OAuth2Application struct {
+		ID                 int64
 		ConfidentialClient bool `xorm:"NOT NULL DEFAULT TRUE"`
 	}
 
-	return x.Sync(new(OAuth2Application))
+	return x.Table("oauth2_application").Sync(new(OAuth2Application))
```
Comment (wxiaoguang) [inline models/migrations/v1_18/v230.go]: Does `Table` really work with `Sync`?

I think the could could be simplified to `type oauth2Application struct`, then no need to use TableName or Table method.

File: models/auth/oauth2_test.go [hunk 3]
```diff
@@ -13,7 +13,12 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-//////////////////// Application
+func TestApplicationTableName(t *testing.T) {
+	e := unittest.GetXORMEngine()
+	tableInfo, err := e.TableInfo(new(auth_model.OAuth2Application))
+	assert.NoError(t, err)
+	assert.EqualValues(t, "oauth2_application", tableInfo.Name)
+}
```
Comment (wxiaoguang) [inline models/auth/oauth2_test.go]: Don't see it really makes sense. It is just testing the XORM's mechanism.

Comment (lunny) [reply models/auth/oauth2_test.go]: 99fef44

File: models/migrations/v1_18/v230.go (hunk 2 above)
Comment (lunny) [reply models/migrations/v1_18/v230.go]: 99fef44

File: models/migrations/v1_18/v230.go [hunk 4]
```diff
@@ -7,11 +7,16 @@ import (
...
+}
+
 // AddConfidentialColumnToOAuth2ApplicationTable: add ConfidentialClient column, setting existing rows to true
 func AddConfidentialClientColumnToOAuth2ApplicationTable(x *xorm.Engine) error {
-	type OAuth2Application struct {
-		ConfidentialClient bool `xorm:"NOT NULL DEFAULT TRUE"`
-	}
-
-	return x.Sync(new(OAuth2Application))
+	return x.Sync(new(oAuth2Application))
```
Comment (wxiaoguang) [inline models/migrations/v1_18/v230.go]: Does this work?

```
func AddConfidentialClientColumnToOAuth2ApplicationTable(x *xorm.Engine) error {
  type oauth2Application struct {
	  ID                 int64
	  ConfidentialClient bool `xorm:"NOT NULL DEFAULT TRUE"`
  }

  return x.Sync(new(oauth2Application))
}
```

File: models/migrations/v1_18/v230_test.go [hunk 5]
```diff
@@ -11,12 +11,16 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func Test_AddConfidentialClientColumnToOAuth2ApplicationTable(t *testing.T) {
-	// premigration
-	type OAuth2Application struct {
-		ID int64
-	}
+// premigration
+type OAuth2Application struct {
```
Comment (wxiaoguang) [inline models/migrations/v1_18/v230_test.go]: Why another ? There is one in `v1_18/v230.go`

Comment (lunny) [reply models/migrations/v1_18/v230_test.go]: They have different fields. This is a part of the tests. At the beginning, there is no newly added column and then add it. So we can not reuse the original struct.

File: models/migrations/v1_18/v230.go (hunk 4 above)
Comment (lunny) [reply models/migrations/v1_18/v230.go]: c41c6d2515c78e52e79d677e85fefb2893cd2393

File: models/migrations/v1_18/v230_test.go (hunk 5 above)
Comment (lunny) [reply models/migrations/v1_18/v230_test.go]: c41c6d2515c78e52e79d677e85fefb2893cd2393

**Review by delvh — APPROVED** [review]

**Pull request merged**